### PR TITLE
Frequency refills

### PR DESCRIPTION
**Pull request by mayerm94**
Add refill and frequency support. Fix the way some attributes were constructing quantities.

Pull requests into CQL QDM Patient API require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] This PR is into the correct branch.
- [x] JIRA ticket for this PR: https://jira.mitre.org/browse/BONNIE-1460
- [x] JIRA ticket links to this PR
- [x] Code diff has been done and been reviewed (it **does not** contain: additional white space, not applicable code changes, debug statements, etc.)
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass (remember to update Gemfile when applicable)
- [x] Code coverage has not gone down and all code touched or added is covered.
    * In rare situations, this may not be possible or applicable to a PR. In those situations:
        1. Note why this could not be done or is not applicable here:
        2. Add TODOs in the code noting that it requires a test
        3. Add a JIRA task to add the test and link it here:
- [x] Automated regression test(s) pass

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA test links: NA
- [x] Justification for using JIRA tests: NA
- [x] JIRA tests have been added to sprint NA

**Reviewer 1:**

Name:
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA tests have been run and pass
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree

**Reviewer 2:**

Name: @losborne
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA tests have been run and pass
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree

### CODE REVIEW

File: app/assets/javascripts/datatypes/medication.js.coffee [hunk 1]
```diff
@@ -34,18 +34,16 @@ class CQL_QDM.MedicationActive extends CQL_QDM.QDMDatatype
...
 
   ###
   @returns {Code}
   ###
   frequency: ->
-    # TODO: Frequency currently not in HDS model.  Can probably add by calculating based off of dose & regimen
-    throw new Error('Bonnie does not currently support MedicationActive.frequency')
     if @_frequency?
-      new cql.Code(@_frequency.code, @_frequency.code_system)
+      new cql.Code(@_frequency.code, @_frequency.code_system, @_frequency.version, @_frequency.title)
```
Comment (jbradl11) [inline app/assets/javascripts/datatypes/medication.js.coffee]: Do we store `version` and `title` for `administrationTiming`?  Will these always be null?

Comment (holmesie) [reply app/assets/javascripts/datatypes/medication.js.coffee]: If they aren't things (per @jbradl11's comment), doing this will *very likely* result in `undefined`, and not `null`. It might be a bit better to do something like:
```
new cql.Code(@_frequency.code, @_frequency.code_system, @_frequency.version || null, @_frequency.title || null)
```

File: spec/javascripts/datatypes/substance_spec.js.coffee [hunk 2]
```diff
@@ -8,19 +8,56 @@ describe "Substance", ->
...
       substanceAdministered = new CQL_QDM.SubstanceAdministered({'start_time': '08/31/2017 1:00 AM', 'end_time': '08/31/2017 2:00 AM'})
-      expect(substanceAdministered.refills).not.toBeDefined()
-      expect(substanceAdministered.frequency).toThrowError('Bonnie does not currently support SubstanceAdministered.frequency')
+      expect(substanceAdministered.frequency()).toEqual null
+      expect(substanceAdministered.supply()).toEqual null
+      expect(substanceAdministered.dosage()).toEqual null
 
   describe "Order", ->
-    it "should throw an error if refills() is called", ->
+    it "should return an integer of the number or refills", ->
```
Comment (losborne) [inline spec/javascripts/datatypes/substance_spec.js.coffee]: 'number or refills' -> 'number of refills'

File: spec/javascripts/datatypes/substance_spec.js.coffee [hunk 3]
```diff
@@ -8,19 +8,56 @@ describe "Substance", ->
...
-      expect(substanceOrder.refills).toThrowError('Bonnie does not currently support SubstanceOrder.refills')
-      expect(substanceOrder.frequency).toThrowError('Bonnie does not currently support SubstanceOrder.frequency')
+      expect(substanceOrder.frequency()).toEqual null
+      expect(substanceOrder.refills()).toEqual null
+      expect(substanceOrder.supply()).toEqual null
+      expect(substanceOrder.dosage()).toEqual null
 
   describe "Recommended", ->
-    it "should throw an error if refills() is called", ->
+    it "should return an integer of the number or refills", ->
```
Comment (losborne) [inline spec/javascripts/datatypes/substance_spec.js.coffee]: 'number or refills' -> 'number of refills'

File: spec/javascripts/datatypes/medication_spec.js.coffee [hunk 4]
```diff
@@ -22,15 +22,29 @@ describe "Medication", ->
       medicationDispensed = new CQL_QDM.MedicationDispensed({'start_time': '08/31/2017 1:00 AM', 'end_time': '08/31/2017 2:00 AM'})
       expect(JSON.stringify(medicationDispensed.relevantPeriod())).toEqual '{"low":"2017-08-31T01:00:00.00+0000","high":"2017-08-31T02:00:00.00+0000","lowClosed":true,"highClosed":true}'
 
-    it "should throw an error if refills() or frequency() is called", ->
+    it "should return an integer of the number or refills", ->
```
Comment (losborne) [inline spec/javascripts/datatypes/medication_spec.js.coffee]: 'number or refills' -> 'number of refills'

File: spec/javascripts/datatypes/medication_spec.js.coffee [hunk 5]
```diff
@@ -51,10 +65,24 @@ describe "Medication", ->
       medicationOrdered = new CQL_QDM.MedicationOrder({'start_time': '08/31/2017 1:00 AM', 'end_time': '08/31/2017 2:00 AM'})
       expect(JSON.stringify(medicationOrdered.relevantPeriod())).toEqual '{"low":"2017-08-31T01:00:00.00+0000","high":"2017-08-31T02:00:00.00+0000","lowClosed":true,"highClosed":true}'
 
-    it "should throw an error if refills() or frequency() is called", ->
+    it "should return an integer of the number or refills", ->
```
Comment (losborne) [inline spec/javascripts/datatypes/medication_spec.js.coffee]: 'number or refills' -> 'number of refills'

File: spec/javascripts/datatypes/medication_spec.js.coffee [hunk 6]
```diff
@@ -74,17 +116,36 @@ describe "Medication", ->
...
+      expect(medicationAdministered.supply()).toEqual null
+      expect(medicationAdministered.dosage()).toEqual null
 
   describe "Discharge", ->
-    it "should throw an error if refills() or frequency() is called", ->
+    it "should return a coded frequency", ->
+      medicationDischarge = new CQL_QDM.MedicationDischarge({'administrationTiming': {code: '1234', code_system: 'SNOMED-CT', version: '', title: 'Test Code'}})
+      expect(medicationDischarge.frequency()).toEqual new cql.Code('1234', 'SNOMED-CT', '', 'Test Code')
+
+    it "should return an integer of the number or refills", ->
```
Comment (losborne) [inline spec/javascripts/datatypes/medication_spec.js.coffee]: 'number or refills' -> 'number of refills'

File: app/assets/javascripts/datatypes/medication.js.coffee (hunk 1 above)
Comment (mayerm94) [reply app/assets/javascripts/datatypes/medication.js.coffee]: Good point -- i'll make the return code match everywhere else that returns a code only using the code and code_system

Comment (mayerm94) [reply app/assets/javascripts/datatypes/medication.js.coffee]: On second thought, the title is non-null on entry. Reverting to include the extra parameters.

Comment (mayerm94) [reply app/assets/javascripts/datatypes/medication.js.coffee]: @holmesie they do result in undefined, which is what they result in if the parameters are not provided (like we do elsewhere) they end up being undefined too
